### PR TITLE
Bluetooth: Controller: Fix elapsed events calculation, peripheral and periodic sync with better abort support

### PR DESCRIPTION
**Pull request by cvinayak**
Introduce BT_CTLR_PERIPHERAL_RESERVE_MAX Kconfig option so
that disabling this option will use minimum time reservation
and exercise the peripheral connection event continuation
using is_abort_cb mechanism.

Introduce BT_CTLR_PERIPHERAL_ISO_RESERVE_MAX Kconfig option
so that disabling this option will use minimum time
reservation and exercise the peripheral ISO connection event
continuation using is_abort_cb mechanism.

Fix BT_CTLR_SCAN_AUX_SYNC_RESERVE_MIN such that event is not
aborted when near supervision timeout conditions.

Fix incorrect elapsed events value when event prepare are
aborted in the pipeline. This caused premature supervision
timeouts.

The PR helps better interoperability with Sony Xperia and Samsung S24 phone under CIS usecases where ACL is placed close after/before CIG events. Also, helps under Broadcast ISO usecases when having Periodic Sync and BIG events close after/before each other.

### CODE REVIEW

**Comment by cvinayak** [comment]
Put to ready to get some PR ready clock time.

This will have conflicts with #78894, and there is pending `is_abort_cb` implementation for CIG events to be added in this PR. But, in general, existing commits in this PR are ready for review.

**Review by Thalley — CHANGES_REQUESTED** [review]
Suggest to do explicit HCI role checks, otherwise LGTM

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn.c [hunk 1]
```diff
@@ -156,6 +170,17 @@ void lll_conn_abort_cb(struct lll_prepare_param *prepare_param, void *param)
...
 	/* NOTE: This is not a prepare being cancelled */
 	if (!prepare_param) {
+		/* Get reference to LLL connection context */
+		lll = param;
+
+		/* For a peripheral role, ensure at least one PDU is tx-ed
+		 * back to central, otherwise let the supervision timeout
+		 * countdown be started.
+		 */
+		if (lll->role && (trx_cnt <= 1U)) {
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn.c]: done

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn.c [hunk 2]
```diff
@@ -171,9 +196,26 @@ void lll_conn_abort_cb(struct lll_prepare_param *prepare_param, void *param)
...
+	/* Get reference to LLL connection context */
 	lll = prepare_param->param;
-	lll->latency_prepare += (prepare_param->lazy + 1);
+
+	/* Accumulate the latency as event is aborted while being in pipeline */
+	lll->lazy_prepare = prepare_param->lazy;
+	lll->latency_prepare += (lll->lazy_prepare + 1U);
+
+#if defined(CONFIG_BT_PERIPHERAL)
+	if (lll->role) {
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn.c]: done

File: subsys/bluetooth/controller/ll_sw/ull_conn_iso.c [hunk 3]
```diff
@@ -493,8 +493,10 @@ void ull_conn_iso_done(struct node_rx_event_done *done)
 							conn->supervision_timeout * 10U * 1000U,
 							cig->iso_interval * CONN_INT_UNIT_US);
 
-				} else if (cis->event_expire > cig->lll.latency_event) {
-					cis->event_expire -= cig->lll.latency_event;
+				} else if (cis->event_expire > (cig->lll.latency_event +
+								cig->lll.lazy_prepare + 1U)) {
+					cis->event_expire -= cig->lll.latency_event +
+							     cig->lll.lazy_prepare + 1U;
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_conn_iso.c]: done

**Review by Thalley — APPROVED** [review]
Not familiar enough with the LL nor the spec to determine if this is correctly implemented, but changes LGTM and it passes CI :D

**Review by kruithofa — APPROVED** [review]

**Pull request merged**